From: Jan Grulich <jgrulich@redhat.com>
Date: Mon, 14 Apr 2025 16:37:00 +0000
Subject: Bug 1954903 - Update libwebrtc GN configs r=webrtc-reviewers,mjf

Update GN config files for the special handling in gn_processor.py for
gbm, drm and PipeWire libraries.

Also remove libwebrtc/third_party/[gbm,drm,pipewire] libs for which we
now have a special handling in gn_processor.py.

Differential Revision: https://phabricator.services.mozilla.com/D242101
Mercurial Revision: https://hg.mozilla.org/mozilla-central/rev/506d4efaf2dc29472623beff64cc2bd0d5555afd
---
 modules/desktop_capture/BUILD.gn | 4 ++++
 modules/portal/BUILD.gn          | 4 +---
 modules/video_capture/BUILD.gn   | 2 +-
 3 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/modules/desktop_capture/BUILD.gn b/modules/desktop_capture/BUILD.gn
index 4e63bd2371..3492abcd87 100644
--- a/modules/desktop_capture/BUILD.gn
+++ b/modules/desktop_capture/BUILD.gn
@@ -555,6 +555,10 @@ rtc_library("desktop_capture") {
       "../../rtc_base:sanitizer",
       "../portal",
     ]
+
+    if (build_with_mozilla) {
+      defines += [ "WEBRTC_USE_PIPEWIRE" ]
+    }
   }
 
   if (rtc_enable_win_wgc) {
diff --git a/modules/portal/BUILD.gn b/modules/portal/BUILD.gn
index 72972d18e1..70b4739ad0 100644
--- a/modules/portal/BUILD.gn
+++ b/modules/portal/BUILD.gn
@@ -139,11 +139,9 @@ if (!build_with_mozilla) {
       deps -= [ ":pipewire_stubs" ]
       defines -= [ "WEBRTC_DLOPEN_PIPEWIRE" ]
       public_deps = [
-        "../../third_party/pipewire",
-        "../../third_party/drm",
-        "../../third_party/gbm",
         "../../third_party/libepoxy"
       ]
+      defines += [ "WEBRTC_USE_PIPEWIRE" ]
     }
   }
 }
diff --git a/modules/video_capture/BUILD.gn b/modules/video_capture/BUILD.gn
index b7acd84839..8a54710d0a 100644
--- a/modules/video_capture/BUILD.gn
+++ b/modules/video_capture/BUILD.gn
@@ -111,7 +111,7 @@ if (!build_with_chromium || is_linux || is_chromeos) {
         ]
         if (build_with_mozilla) {
           configs -= [ "../portal:pipewire_base" ]
-          public_deps = [ "../../third_party/pipewire" ]
+          defines = [ "WEBRTC_USE_PIPEWIRE" ]
         }
       }
     }
